extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -136,13 +151,23 @@ export class AmpAdExit extends AMP.BaseElement {
             const vendorResponse = replacements./*OK*/expandStringSync(
                 customVar.iframeTransportSignal, {
                   'IFRAME_TRANSPORT_SIGNAL': (vendor, responseKey) => {
+                    if (!(vendor && responseKey)) {
```

Comment:
What does vendor represent here? I it's a string, but why is it necessary?

Reply:
See opening section of the design doc that I linked to in my last comment. It's basically a company like MOAT, etc. which creates a non-friendly iframe (which can thus run arbitrary JS) to receive amp-analytics signals, and can then choose to send a response. The response is a JSON object, but the keys/values must be strings (i.e. depth can only be 1). Then what's going on here in amp-ad-exit is that we're subbing those values into the ad's exit URL.
somePage.html?foo=_bar -> look in the config to see that _bar = "IFRAME_TRANSPORT_SIGNAL(example-3p-vendor,collected-data)" -> the vendor 'example-3p-vendor' sent an object like {collected-data: abc, favoriteColor: blue } -> we send them to somePage.html?foo=abc

Reply:
(The list of possible vendors = the keys of the huge map in extensions/amp-analytics/0.1/vendors.js...but we are limiting this to MRC-accredited vendors, so realistically there will be a single-digit number of eligible vendors for the foreseeable future.)

Reply:
It doesn't look like we need multiple vendors per ad-exit instance, though? So it could just be `IFRAME_TRANSPORT_SIGNAL(varName)`?

Reply:
We do not limit to a single 3p amp-analytics vendor per creative, nor per ad-exit instance. So to do that would create the potential for naming collisions.